src/main/java/hudson/plugins/emailext/plugins/ContentBuilder.java

```diff
@@ -68,7 +68,9 @@ public static String transformText(String origText, ExtendedEmailPublisherContex
             List<TokenMacro> macros = new ArrayList<TokenMacro>(getPrivateMacros());
             if(additionalMacros != null)
                 macros.addAll(additionalMacros);
-            newText = TokenMacro.expandAll(context.getBuild(), context.getListener(), newText, false, macros);
+            if(context.getBuild() != null) {
+                newText = TokenMacro.expandAll(context.getBuild(), context.getListener(), newText, false, macros);

```

Comment:
As i understand that wouldn't work and i also depend on `TokenMacro.expandAll(build, ...` method. First of all TokenMacro + workflow should be resolved imho.